src/receive_imf.rs

```diff
@@ -1873,34 +1873,40 @@ async fn lookup_chat_or_create_adhoc_group(
     if !contact_ids.contains(&from_id) {
         contact_ids.push(from_id);
     }
-    if let Some((chat_id, blocked)) = context
-        .sql
-        .query_row_optional(
-            &format!(
+    let trans_fn = |t: &mut rusqlite::Transaction| {
+        t.execute(
+            "CREATE TABLE contacts_tmp (
```

Comment:
Not good that this became a write transaction, but the function call is expected to be heavy anyway if the chat isn't found and an ad-hoc group needs to be created